infra/tf/variables.tf

```diff
@@ -0,0 +1,106 @@
+################################################################################
+# ZQ2 GCP Terraform input variables
+################################################################################
+
+
+variable "project_id" {
+  description = "Project ID used to dpeloy the ZQ2 network"
+  type        = string
+  nullable    = false
+}
+
+variable "region" {
+  description = "The region ID to host the network in"
+  type        = string
+  default     = "europe-west2"
+}
+
+variable "node_zone" {
+  type     = string
+  default  = "europe-west2-a"
+  nullable = false
+}
+
+variable "eth_chain_id" {
+  description = "ZQ2 EVM Chain ID"
+  type        = number
+  nullable    = false
+  default     = 33469
+}
+
+variable "subdomain" {
+  type     = string
+  nullable = false
+}
+
+variable "network_name" {
+  description = "(Optional) ZQ2 network name"
+  type        = string
+  nullable    = false
+  default     = "zq2-devnet"
+}
+
+variable "labels" {
+  type        = map(string)
+  description = "A single-level map/object with key value pairs of metadata labels to apply to the GCP resources. All keys should use underscores and values should use hyphens. All values must be wrapped in quotes."
+  nullable    = true
+  default     = {}
+}
+
+variable "zq2_type" {
```

Comment:
Maybe a style choice, but IMO this is a bit of an anti-pattern. I don't think we should specify a 'type' which controls how different resources are configured. I think we should expose those resource configurations more directly.

Reply:
same as below - backward compatibility with the in use provisioning. Can be removed.